### PR TITLE
drivers/video/fb: add poll support

### PR DESCRIPTION
**Pull request by FASTSHIFT**
## Summary
Add poll method for framebuffer device to achieve more lightweight vsync synchronization.

## Impact
Applications using framebuffer devices.

## Testing
Daliy test.

### CODE REVIEW

**Comment by xiaoxiang781216** [comment]
@FASTSHIFT please squash into one patch

**Comment by FASTSHIFT** [comment]
> @FASTSHIFT please squash into one patch

done

File: drivers/video/fb.c [hunk 1]
```diff
@@ -544,6 +548,78 @@ static int fb_ioctl(FAR struct file *filep, int cmd, unsigned long arg)
...
+ *   Zero (OK) is returned success; a negated errno value is returned on any
+ *   failure.
+ *
+ ****************************************************************************/
+
+int fb_pollnotify(FAR struct fb_vtable_s *vtable)
+{
+  DEBUGASSERT(vtable != NULL);
+
+  if (vtable->chardev == NULL || vtable->chardev->fds == NULL)
```
Comment (xiaoxiang781216) [inline drivers/video/fb.c:611]: remove the check don't need

File: drivers/video/fb.c [hunk 2]
```diff
@@ -544,6 +548,78 @@ static int fb_ioctl(FAR struct file *filep, int cmd, unsigned long arg)
...
+ * Input Parameters:
+ *   vtable - Pointer to framebuffer's virtual table.
+ *
+ * Returned Value:
+ *   Zero (OK) is returned success; a negated errno value is returned on any
+ *   failure.
+ *
+ ****************************************************************************/
+
+int fb_pollnotify(FAR struct fb_vtable_s *vtable)
```
Comment (xiaoxiang781216) [inline drivers/video/fb.c:607]: int->void

File: include/nuttx/video/fb.h [hunk 3]
```diff
@@ -613,6 +615,10 @@ struct fb_vtable_s
   /* Enable/disable panel power (0: full off). */
 
   int (*setpower)(FAR struct fb_vtable_s *vtable, int power);
+
+  /* Pointer to framebuffer device private data. */
+
+  FAR struct fb_chardev_s *chardev;
```
Comment (xiaoxiang781216) [inline include/nuttx/video/fb.h:621]:
```suggestion
  FAR struct void *priv;
```
and remove line 299

File: drivers/video/fb.c (hunk 1 above)
Comment (FASTSHIFT) [reply drivers/video/fb.c]: done

File: drivers/video/fb.c (hunk 2 above)
Comment (FASTSHIFT) [reply drivers/video/fb.c]: done

File: include/nuttx/video/fb.h (hunk 3 above)
Comment (FASTSHIFT) [reply include/nuttx/video/fb.h]: done

File: drivers/video/fb.c [hunk 4]
```diff
@@ -544,6 +548,79 @@ static int fb_ioctl(FAR struct file *filep, int cmd, unsigned long arg)
...
+{
+  FAR struct fb_chardev_s *fb;
+
+  DEBUGASSERT(vtable != NULL);
+
+  fb = vtable->priv;
+
+  /* If there is no listener, it is not processed. */
+
+  if (fb->fds == NULL)
```
Comment (xiaoxiang781216) [inline drivers/video/fb.c:612]: don't need since poll_notify check it internally

Comment (FASTSHIFT) [reply drivers/video/fb.c]: done

**Review by jerpelea — APPROVED** [review]

**Pull request merged**